esx_service/utils/auth_api.py

```diff
@@ -457,7 +457,7 @@ def _tenant_create(name, default_datastore, description="", vm_list=None, privil
 
     # VM list can be empty during tenant create. Validate only if it exists
     vms = None
-    if vm_list:
+    if vm_list and name != auth_data_const.DEFAULT_TENANT:
```

Comment:
I think we should error out here when user give non-empty vm_list while creating DEFAULT_TENANT. We should not ignore the vm_list.

I think similar check should also apply with "tenant vm add", "tenant vm rm" and "tenant vm replace".